Invalid JSON element for number: "8"

To reproduce:

Click Patient registration example.

Error:

2021-01-24 14:26:56.251 13762-13762/com.google.android.fhir.datacapture.gallery E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.google.android.fhir.datacapture.gallery, PID: 13762
    java.lang.RuntimeException: An exception happened in constructor of class com.google.android.fhir.datacapture.QuestionnaireViewModel
        at androidx.lifecycle.SavedStateViewModelFactory.create(SavedStateViewModelFactory.java:132)
        at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.java:185)
        at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.java:150)
        at androidx.lifecycle.ViewModelLazy.getValue(ViewModelProvider.kt:54)
        at androidx.lifecycle.ViewModelLazy.getValue(ViewModelProvider.kt:41)
        at com.google.android.fhir.datacapture.QuestionnaireFragment.getViewModel(Unknown Source:2)
        at com.google.android.fhir.datacapture.QuestionnaireFragment.onViewCreated(QuestionnaireFragment.kt:53)
        at androidx.fragment.app.Fragment.performViewCreated(Fragment.java:2974)
        at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:538)
        at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:278)
        at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:2177)
        at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:2088)
        at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1990)
        at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:3122)
        at androidx.fragment.app.FragmentManager.dispatchActivityCreated(FragmentManager.java:3056)
        at androidx.fragment.app.FragmentController.dispatchActivityCreated(FragmentController.java:251)
        at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:473)
        at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:201)
        at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1425)
        at android.app.Activity.performStart(Activity.java:7825)
        at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3294)
        at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:221)
        at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:201)
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:173)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2016)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
     Caused by: java.lang.IllegalArgumentException: Error parsing field: google.fhir.r4.core.Extension.ValueX.integer
        at com.google.fhir.common.JsonFormat$Parser.parseFieldValue(JsonFormat.java:957)
        at com.google.fhir.common.JsonFormat$Parser.mergeField(JsonFormat.java:857)
        at com.google.fhir.common.JsonFormat$Parser.mergeChoiceField(JsonFormat.java:826)
        at com.google.fhir.common.JsonFormat$Parser.mergeMessage(JsonFormat.java:762)
        at com.google.fhir.common.JsonFormat$Parser.parseFieldValue(JsonFormat.java:971)
        at com.google.fhir.common.JsonFormat$Parser.mergeRepeatedField(JsonFormat.java:897)
        at com.google.fhir.common.JsonFormat$Parser.mergeField(JsonFormat.java:855)
        at com.google.fhir.common.JsonFormat$Parser.mergeMessage(JsonFormat.java:764)
        at com.google.fhir.common.JsonFormat$Parser.parseFieldValue(JsonFormat.java:971)
        at com.google.fhir.common.JsonFormat$Parser.mergeRepeatedField(JsonFormat.java:897)
        at com.google.fhir.common.JsonFormat$Parser.mergeField(JsonFormat.java:855)
        at com.google.fhir.common.JsonFormat$Parser.mergeMessage(JsonFormat.java:764)
        at com.google.fhir.common.JsonFormat$Parser.parseFieldValue(JsonFormat.java:971)
2021-01-24 14:26:56.251 13762-13762/com.google.android.fhir.datacapture.gallery E/AndroidRuntime:     at com.google.fhir.common.JsonFormat$Parser.mergeRepeatedField(JsonFormat.java:897)
        at com.google.fhir.common.JsonFormat$Parser.mergeField(JsonFormat.java:855)
        at com.google.fhir.common.JsonFormat$Parser.mergeMessage(JsonFormat.java:764)
        at com.google.fhir.common.JsonFormat$Parser.merge(JsonFormat.java:680)
        at com.google.fhir.common.JsonFormat$Parser.merge(JsonFormat.java:691)
        at com.google.android.fhir.datacapture.QuestionnaireViewModel.<init>(QuestionnaireViewModel.kt:40)
        at java.lang.reflect.Constructor.newInstance0(Native Method)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
        at androidx.lifecycle.SavedStateViewModelFactory.create(SavedStateViewModelFactory.java:122)
        	... 31 more
     Caused by: java.lang.IllegalArgumentException: Invalid JSON element for number: "8"
        at com.google.fhir.wrappers.PrimitiveWrappers.checkIsNumber(PrimitiveWrappers.java:204)
        at com.google.fhir.wrappers.PrimitiveWrappers.parseAndWrap(PrimitiveWrappers.java:155)
        at com.google.fhir.common.JsonFormat$Parser.parseFieldValue(JsonFormat.java:953)
        	... 52 more

--- a/datacapturegallery/src/main/assets/patient-registration.json
+++ b/datacapturegallery/src/main/assets/patient-registration.json
@@ -14,11 +14,8 @@
           "text": "ANC ID",
           "type": "string",
           "required": true,
+          "maxLength": 8,
           "extension": [
-            {
-              "url": "http://hl7.org/fhir/StructureDefinition/questionnaire-maxLength",
-              "valueInteger": 8
-            },
             {
               "url": "http://hl7.org/fhir/StructureDefinition/questionnaire-instruction",
               "valueString": "Unique ANC ID"
@@ -30,11 +27,8 @@
           "text": "First name",
           "type": "string",
           "required": true,
+          "maxLength": 30,
           "extension": [
-            {
-              "url": "http://hl7.org/fhir/StructureDefinition/questionnaire-maxLength",
-              "valueInteger": 30
-            },
             {
               "url": "http://hl7.org/fhir/StructureDefinition/questionnaire-instruction",
               "valueString": "Given name"
@@ -46,11 +40,8 @@
           "text": "Last name",
           "type": "string",
           "required": true,
+          "maxLength": 50,
           "extension": [
-            {
-              "url": "http://hl7.org/fhir/StructureDefinition/questionnaire-maxLength",
-              "valueInteger": 50
-            },
             {
               "url": "http://hl7.org/fhir/StructureDefinition/questionnaire-instruction",
               "valueString": "Family name"
@@ -87,11 +78,8 @@
           "linkId": "PR-6",
           "text": "Mobile number",
           "type": "integer",
+          "maxLength": 8,
           "extension": [
-            {
-              "url": "http://hl7.org/fhir/StructureDefinition/questionnaire-maxLength",
-              "valueInteger": 8
-            },
             {
               "url": "http://hl7.org/fhir/StructureDefinition/questionnaire-instruction",
               "valueString": "Mobile phone number"
@@ -114,11 +102,8 @@
           "text": "Alternate contact name",
           "type": "string",
           "required": true,
+          "maxLength": 50,
           "extension": [
-            {
-              "url": "http://hl7.org/fhir/StructureDefinition/questionnaire-maxLength",
-              "valueInteger": 50
-            },
             {
               "url": "http://hl7.org/fhir/StructureDefinition/questionnaire-instruction",
               "valueString": "Alternate contact name"
@@ -129,11 +114,8 @@
           "linkId": "PR-10",
           "text": "Alternate contact number",
           "type": "integer",
+          "maxLength": 8,
           "extension": [
-            {
-              "url": "http://hl7.org/fhir/StructureDefinition/questionnaire-maxLength",
-              "valueInteger": 8
-            },
             {
               "url": "http://hl7.org/fhir/StructureDefinition/questionnaire-instruction",
               "valueString": "Phone number for alternate contact"

